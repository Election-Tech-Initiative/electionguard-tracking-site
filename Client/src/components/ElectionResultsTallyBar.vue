<template>
  <div id="bar" class="bar-wrapper" :class="{'first': isFirst}">
    <div class="bar" :style="{'width': percentage}"></div>
    <div
      class="bar-text"
      :style="{'min-width': percentage}"
    >{{count}} ({{percentage}})</div>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

@Component
export default class ElectionResultsTallyBar extends Vue {
  @Prop() public isFirst!: boolean;
  @Prop() public count!: string;
  @Prop() public percentage!: string;
}
</script>

<style lang="scss" scoped>
.first {
  font-weight: bold;

  .bar {
    background-color: $primary;
  }
  .bar-text {
    color: $white;
  }
}

.bar-wrapper {
  position: relative;
}

.bar {
  min-width: 0.1rem;
  height: 2rem;
  background-color: $white-ter;
}

.bar-text {
  display: inline-block;
  position: absolute;
  left: 0;
  top: 0;
  text-align: right;
  padding: 0.2rem 0.6rem;
  text-shadow: 0.05rem 0.05rem 0.1rem $grey-darker;
}
</style>
