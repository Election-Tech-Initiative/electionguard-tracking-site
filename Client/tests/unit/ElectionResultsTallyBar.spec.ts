import { mount } from '@vue/test-utils';

import ElectionResultsTallyBar from '@/components/ElectionResultsTallyBar.vue';

describe('ElectionResultsTally.vue', () => {
    const getMountedWrapper = (isFirst: boolean) => mount(ElectionResultsTallyBar, {
        mocks: {
            $t: (msg: string): string => msg
        },
        propsData: {
            isFirst,
            count: '5',
            percentage: '50%',
        },
    });

    it('renders with first class', () => {
        const wrapper = getMountedWrapper(true);
        expect(wrapper.element).toMatchSnapshot('with first class');
    });

    it('renders without first class', () => {
        const wrapper = getMountedWrapper(false);
        expect(wrapper.element).toMatchSnapshot('without first class');
    });
});
