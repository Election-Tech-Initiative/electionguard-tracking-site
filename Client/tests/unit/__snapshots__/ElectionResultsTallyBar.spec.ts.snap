// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ElectionResultsTally.vue renders with first class: with first class 1`] = `
<div
  class="bar-wrapper first"
  id="bar"
>
  <div
    class="bar"
    style="width: 50%;"
  />
   
  <div
    class="bar-text"
    style="min-width: 50%;"
  >
    5 (50%)
  </div>
</div>
`;

exports[`ElectionResultsTally.vue renders without first class: without first class 1`] = `
<div
  class="bar-wrapper"
  id="bar"
>
  <div
    class="bar"
    style="width: 50%;"
  />
   
  <div
    class="bar-text"
    style="min-width: 50%;"
  >
    5 (50%)
  </div>
</div>
`;
